### DISCUSSION TITLE
Platform projects for Mac and Windows ?

### DISCUSSION THREAD

**Discussion by guldud** [AvaloniaUI/Avalonia#14824] [Q&A] [answered]
Is there a way or a template to create specific platform project for Mac or Windows like we know from Android and iOS ? Or even the runner projects from Flutter? 

The app.Desktop approach is convenient, but I'm looking for a way to access specific OS features, modify Mac OS X .plist and the like.

**Comment by timunie** [comment]
Should be easy to do on your own by copying what Desktop project does.

**Reply by guldud** [reply]
Does anyone know of an example? For example, to implement some Interface on Mac OS X, and inject it into the main class library ?

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/avalonia-docs/blob/379d6a3095230c4bc8423b7b27c1f4bcce25f29c/docs/guides/implementation-guides/how-to-implement-dependency-injection.md

**Reply by guldud** [reply]
I do know how injection work. What I would like to see is how to - for example - construct a platform project that looks like the one for iOS but for Mac OS X instead. A Mac OS X project that have a main.cs which uses UIKit (or AppKit). A project that have a AppDelegate.cs, a info.plist file, and so on.

**Comment by maxkatz6** [comment]
Yes, you can. Simply copy-pasting the same project works, since most of shared code is already extracted to the common project.
But you also can change TargetFramework to "net8.0-window" and "net8.0-macos" if needed, it will give you more access to the platform APIs.

**Reply by guldud** [reply]
Ah. I get it now. Thanks @timunie and @maxkatz6

**Reply by guldud** [reply]
It is not as straightforward as I initially assumed. 
I copied the desktop project. 
Changed the target framework to net8.0-macos
Changed the output type to Exe
Added a reference info.plist file from https://docs.avaloniaui.net/docs/deployment/macOS#manual
Change various keys, like CFBundleIdentifier to a proper values
..and tried to build.

I'm getting 

9>Xamarin.Shared.targets(612,3): Error  : A bundle identifier is required. Either add an 'ApplicationId' property in the project file, or add a 'CFBundleIdentifier' entry in the project's Info.plist file.
9>------- Finished building project: MyMacApp.Mac. Succeeded: False. Errors: 1. Warnings: 0

Seems odd with a Xamarin error. And the bundle identifier has been supplied. 

I'm gonna dig a little bit deeper to see if I can find out how to construct a proper Mac project

**Reply by maxkatz6** [reply, marked as answer]
Please ignore https://docs.avaloniaui.net/docs/deployment/macOS#manual page, if you are building with "net8.0-macos", as it does ".app" packaging job for you. Simple "dotnet publish" is enough there. And CFBundleIdentifier is ignored, you need to set [ApplicationId](https://github.com/xamarin/xamarin-macios/blob/bb775d026c72991652ce7254dcf116125b848b38/dotnet/SingleProject.md?plain=1#L15-L18).

**Reply by guldud** [reply]
It worked! I can build. I can publish. I got .app. Life is great.

**Reply by guldud** [reply]
Why does it still need app.manifest though ? For a net8.0-macos project?

**Reply by maxkatz6** [reply]
`app.manifest` is only needed for Windows project. No need for macOS.

**Reply by guldud** [reply]
You're right. My bad. A faulty entry in the csproj.
